### PR TITLE
fix buying from loot actor adding wrong amount

### PR DESCRIPTION
**Pull request by Stexinator**
refactor loot actor
refactor itemMixin -> addItem, removeItem

### CODE REVIEW

File: module/actor/witcherActor.js [hunk 1]
```diff
@@ -493,6 +493,20 @@ export default class WitcherActor extends Actor {
         );
     }
 
+    async addItem(addItem, numberOfItem, forcecreate = false) {
+        let foundItem = this.items.find(item => item.name == addItem.name && item.type == addItem.type);
+        if (foundItem && !forcecreate && !foundItem.system.isStored) {
+            await foundItem.update({ 'system.quantity': Number(foundItem.system.quantity) + Number(numberOfItem) });
+        } else {
+            let newItem = { ...addItem.toObject() };
```
Comment (Stexinator) [inline module/actor/witcherActor.js:501]: @SquidOfDoomed the ``toObject`` here is the solution

Comment (SquidOfDoomed) [reply module/actor/witcherActor.js:501]: You mean to make it an object, then adjust it and then to call addItem?

Comment (Stexinator) [reply module/actor/witcherActor.js:501]: When making the copy, call toObject of the original

**Comment by SquidOfDoomed** [comment]
@Stexinator I noticed that with your change it is also now possible to buy 0 items. This would not make sense I think. It is also possible to buy 0.1 of an item. Interesting but probably not intended. I like the now gone "other" default.

Another thing I just noticed is that some parts are just showing commas where others show dots. Also it is not possible to buy 1,1 items but just 1.1 items.
image

Would it make sense to fix this together in your refactoring or seperately?

**Comment by Stexinator** [comment]
There is only so much energy I want to invest to prevent people from doing stupid stuff

**Pull request merged**